Mathlib/Computability/DFA.lean

```diff
@@ -11,11 +11,26 @@ import Mathlib.Tactic.NormNum
 /-!
 # Deterministic Finite Automata
 
-This file contains the definition of a Deterministic Finite Automaton (DFA), a state machine which
-determines whether a string (implemented as a list over an arbitrary alphabet) is in a regular set
-in linear time.
-Note that this definition allows for Automaton with infinite states, a `Fintype` instance must be
-supplied for true DFA's.
+A Deterministic Finite Automaton (DFA) is a state machine which
+decides membership in a particular regular language, by following a path
+uniquely determined by an input string.
+
+We define regular languages to be ones for which a DFA exists, other formulations
+are then proved equivalent.
+
```

Comment:
```suggestion
A Deterministic Finite Automaton (DFA) is a state machine which
decides membership in a particular language, by following a path
uniquely determined by an input string.

We define regular languages to be ones for which a DFA exists, other formulations
are then proved equivalent.

```
This new docstring is an improvement, but my suggestion makes it clearer that this isn't a circular definition!
I think it'd also be nice to preserve the "string (implemented as a list over an arbitrary alphabet)" bit, since `String` in Lean can mean something else, and the list clarification is helpful.